### PR TITLE
fix(vpc_gw_dhcp): Properly set fields that should default to true

### PR DESCRIPTION
**Pull request by Tuetuopay**
Terraform will make sure that fields not present in the file will be set, so we never send nil to the api and get its defaults. Set the defaults ourselves to be consistent with the docs.

Intended to fix https://github.com/scaleway/terraform-provider-scaleway/issues/1059

### CODE REVIEW

File: scaleway/resource_vpc_public_gateway_dhcp_test.go [hunk 1]
```diff
@@ -53,6 +53,17 @@ func testAccCheckScalewayVPCPublicGatewayDHCPExists(tt *TestTools, n string) res
 			return err
 		}
 
+		// Test default values
+		if !dhcp.EnableDynamic {
+			return fmt.Errorf("enable_dynamic is false, should defautl to true")
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway_dhcp_test.go:58]:
```suggestion
			return fmt.Errorf("enable_dynamic is false, should default to true")
```

File: scaleway/resource_vpc_public_gateway_dhcp_test.go [hunk 2]
```diff
@@ -53,6 +53,17 @@ func testAccCheckScalewayVPCPublicGatewayDHCPExists(tt *TestTools, n string) res
...
+		}
+		if !dhcp.PushDefaultRoute {
+			return fmt.Errorf("push_default_route is false, should defautl to true")
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway_dhcp_test.go:61]:
```suggestion
			return fmt.Errorf("push_default_route is false, should default to true")
```

File: scaleway/resource_vpc_public_gateway_dhcp_test.go [hunk 3]
```diff
@@ -53,6 +53,17 @@ func testAccCheckScalewayVPCPublicGatewayDHCPExists(tt *TestTools, n string) res
...
+		}
+		if !dhcp.PushDNSServer {
+			return fmt.Errorf("push_dns_server is false, should defautl to true")
```
Comment (remyleone) [inline scaleway/resource_vpc_public_gateway_dhcp_test.go:64]:
```suggestion
			return fmt.Errorf("push_dns_server is false, should default to true")
```

File: scaleway/resource_vpc_public_gateway_dhcp_test.go (hunk 1 above)
Comment (Tuetuopay) [reply scaleway/resource_vpc_public_gateway_dhcp_test.go]: fixed

File: scaleway/resource_vpc_public_gateway_dhcp_test.go (hunk 3 above)
Comment (Tuetuopay) [reply scaleway/resource_vpc_public_gateway_dhcp_test.go]: fixed

File: scaleway/resource_vpc_public_gateway_dhcp_test.go (hunk 2 above)
Comment (Tuetuopay) [reply scaleway/resource_vpc_public_gateway_dhcp_test.go]: fixed

**Review by Monitob — DISMISSED** [review]

File: scaleway/resource_vpc_public_gateway_dhcp.go [hunk 4]
```diff
@@ -53,8 +53,8 @@ func resourceScalewayVPCPublicGatewayDHCP() *schema.Resource {
 			},
 			"enable_dynamic": {
 				Type:        schema.TypeBool,
-				Computed:    true,
 				Optional:    true,
+				Default:     true,
```
Comment (Monitob) [inline scaleway/resource_vpc_public_gateway_dhcp.go:57]: This has an impact on the pat-rules test. You should just rerun the test to update the cassettes

**Review by Monitob — CHANGES_REQUESTED** [review]

**Pull request merged**